locations/middlewares/anti_bot_detection.py

```diff
@@ -0,0 +1,86 @@
+from enum import Enum
+
+from scrapy import Request, Spider
+from scrapy.downloadermiddlewares.httpcompression import HttpCompressionMiddleware
+from scrapy.http import HtmlResponse, Response
+
+
+# As a temporary transition to use of AntiBotDetectionMiddleware,
+# assume that Zyte API is capable of bypassing all anti-bot
+# methods. This list will need to be amended after testing of Zyte
+# API ability to bypass various anti-bot methods. This list will
+# need to be frequently updated depending on the effectiveness of
+# Zyte API's ability to rapidly bypass new/revised anti-bot methods.
+class AntiBotMethods(Enum):
+    AZURE_WAF = {"name": "Azure WAF", "zyte_bypassable": True}
+    CLOUDFLARE = {"name": "Cloudflare", "zyte_bypassable": True}
+    DATADOME = {"name": "DataDome", "zyte_bypassable": True}
+    HUMAN = {"name": "HUMAN", "zyte_bypassable": True}
+    IMPERVA = {"name": "Imperva", "zyte_bypassable": True}
+    QRATOR = {"name": "Qrator", "zyte_bypassable": True}
```

Comment:
If it's easy, can you add comments with links to documentation about these systems?

Reply:
Can do, it's easy :)

Reply:
Friendly nudge on this.